solc/CommandLineParser.cpp

```diff
@@ -469,11 +472,26 @@ void CommandLineParser::parseOutputSelection()
 		solAssert(false, "");
 	};
 
+	int numberOfSelectedOutputs = (m_args.count(g_strGas) > 0) ? 1 : 0;
 	for (auto&& [optionName, outputComponent]: CompilerOutputs::componentMap())
+	{
 		m_options.compiler.outputs.*outputComponent = (m_args.count(optionName) > 0);
+		if (m_args.count(optionName) > 0)
+			numberOfSelectedOutputs++;
+		if (numberOfSelectedOutputs > 1 && (m_args.count(g_strMachineReadable) > 0))
+			solThrow(
+				CommandLineValidationError,
+				"Option --" + g_strMachineReadable + " can't be used with more than one output selected."
+			);
+	}
 
 	if (m_options.input.mode == InputMode::Assembler && m_options.compiler.outputs == CompilerOutputs{})
 	{
+		if (m_options.formatting.machineReadable)
+			solThrow(
+				CommandLineValidationError,
+				"Please select only one output option when using  --" + g_strMachineReadable + "."
+			);
```

Comment:
It would be nice if we could be more user-friendly here and show all the output options that were selected. User might not be aware for example that `--gas` counts as output.

Actually maybe `--gas` should be grouped with other options, but that's a separate discussion.